core/Inst.hpp

```diff
@@ -362,8 +422,9 @@ namespace olympia
         sparta::memory::addr_t target_vaddr_ =
             0; // Instruction's Target PC (for branches, loads/stores)
         bool is_oldest_ = false;
-        uint64_t unique_id_ = 0;      // Supplied by Fetch
-        uint64_t program_id_ = 0;     // Supplied by a trace Reader or execution backend
+        uint64_t unique_id_ = 0;  // Supplied by Fetch
+        uint64_t program_id_ = 0; // Supplied by a trace Reader or execution backend
+        uint64_t uopid_ = 0;      // Set in decode
```

Comment:
If you used Sparta's `ValidValue` class here you wouldn't need `bool has_uops_`.

Reply:
Addressed in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)